### PR TITLE
Preserve internal state if disconnected from MQTT Broker

### PR DESCRIPTION
**Pull request by OleksandrChaika**

### CODE REVIEW

File: src/Mqtt/MqttProcessor.cpp [hunk 1]
```diff
@@ -126,20 +126,20 @@ namespace {
                     ctx.desiredState.activityId = c.activityId;
                     ctx.stateMachine->handleEvent(c.event);
```
Comment (OlavHerbst) [inline src/Mqtt/MqttProcessor.cpp:127]: If all handled exceptions occur in this statement, then move the try-catch-block to here.

File: src/Mqtt/MqttProcessor.cpp [hunk 2]
```diff
@@ -126,20 +126,20 @@ namespace {
...
-                    sua::Logger::error("Invalid topic: {}", topic);
+                    sua::Logger::error("Invalid topic: '{}'", topic);
                 }
             } catch(const nlohmann::json::parse_error & e) {
-                sua::Logger::error("Invalid request for {}, unable to parse json: {}", topic, e.what());
+                sua::Logger::error("Unable to parse desired state request (not a valid json?): '{}'", e.what());
             } catch(const nlohmann::json::exception& e) {
                 ctx.desiredState.activityId = nlohmann::json::parse(message).at("activityId");
-                sua::Logger::error("Invalid request for {}: {}", topic, e.what());
+                sua::Logger::error("Incomplete desired state request, unable to identify: '{}'", e.what());
```
Comment (OlavHerbst) [inline src/Mqtt/MqttProcessor.cpp]: mention "json" in error message

File: src/Mqtt/MqttProcessor.cpp [hunk 3]
```diff
@@ -126,20 +126,20 @@ namespace {
...
                 mqtt->send(sua::IMqttProcessor::TOPIC_FEEDBACK, sua::MqttMessage::Identifying);
                 mqtt->send(sua::IMqttProcessor::TOPIC_FEEDBACK, sua::MqttMessage::IdentificationFailed, e.what());
             } catch(const std::logic_error & e) {
-                sua::Logger::error("Invalid request for {}: {}", topic, e.what());
+                sua::Logger::error("Invalid desired state request: '{}'", e.what());
             } catch(const std::runtime_error & e) {
                 ctx.desiredState.activityId = nlohmann::json::parse(message).at("activityId");
-                sua::Logger::error("Invalid request for {}: {}", topic, e.what());
+                sua::Logger::error("Malformed desired state request, unable to identify: '{}'", topic, e.what());
```
Comment (OlavHerbst) [inline src/Mqtt/MqttProcessor.cpp]: remove topic

File: src/Mqtt/MqttProcessor.cpp [hunk 4]
```diff
@@ -126,20 +126,20 @@ namespace {
...
                 ctx.desiredState.activityId = nlohmann::json::parse(message).at("activityId");
-                sua::Logger::error("Invalid request for {}: {}", topic, e.what());
+                sua::Logger::error("Incomplete desired state request, unable to identify: '{}'", e.what());
                 mqtt->send(sua::IMqttProcessor::TOPIC_FEEDBACK, sua::MqttMessage::Identifying);
                 mqtt->send(sua::IMqttProcessor::TOPIC_FEEDBACK, sua::MqttMessage::IdentificationFailed, e.what());
             } catch(const std::logic_error & e) {
-                sua::Logger::error("Invalid request for {}: {}", topic, e.what());
+                sua::Logger::error("Invalid desired state request: '{}'", e.what());
             } catch(const std::runtime_error & e) {
                 ctx.desiredState.activityId = nlohmann::json::parse(message).at("activityId");
```
Comment (OlavHerbst) [inline src/Mqtt/MqttProcessor.cpp:147]: Why is activityId extracted if runtime_error or json::exception, but not at the other exceptions? Missing a comment.

**Review by OlavHerbst — CHANGES_REQUESTED** [review]
error in error message

File: src/Mqtt/MqttProcessor.cpp (hunk 1 above)
Comment (OleksandrChaika) [reply src/Mqtt/MqttProcessor.cpp:127]: Not only here, also in ->read... call to protocol parser.

File: src/Mqtt/MqttProcessor.cpp (hunk 2 above)
Comment (OleksandrChaika) [reply src/Mqtt/MqttProcessor.cpp]: JSON added to the message

File: src/Mqtt/MqttProcessor.cpp (hunk 3 above)
Comment (OleksandrChaika) [reply src/Mqtt/MqttProcessor.cpp]: Removed

File: src/Mqtt/MqttProcessor.cpp (hunk 4 above)
Comment (OleksandrChaika) [reply src/Mqtt/MqttProcessor.cpp:147]: Comments added. But exception handling can be improved definitely. I think we could parse request as it is and then do a consistency check after that.

**Review by OlavHerbst — APPROVED** [review]
ok

File: src/Mqtt/MqttProcessor.cpp (hunk 4 above)
Comment (OlavHerbst) [reply src/Mqtt/MqttProcessor.cpp:147]: ok

**Pull request merged**